passenger-rails/2.3/nginx.conf.erb

```diff
@@ -56,9 +56,15 @@ http {
   # Performance Helps
   sendfile on;
 
+  # Intercept requests whose hosts don't match the WHITELISTED_HOSTS and return a 400
+  server {
+    listen 80 default_server;
+    return 400 "Bad Request";
+  }
+
   server {
     listen 80;
-    server_name default_server;
+    server_name <%= ENV.fetch('WHITELISTED_HOSTS', '.youearnedit.com') %>;
```

Comment:
Yes. This basically accepts any request to *.youearnedit.com or whatever configured domain names and rejects them with a 400 if the host header is anything else. That's the only way I can figure to handle the subdomains since nginx won't allow multiple if statements.

Reply:
WHITELISTED_HOSTS would get passed in when the container runs if the value needed to be changed. It defaults to .youearnedit.com.